### DISCUSSION TITLE
How to disable agent report data to collector at runtime/by code

### DISCUSSION THREAD

**Discussion by lag1500** [apache/skywalking#8017] [General]
For example, I want to disable report on 10.1.2.3 (of course IP condition is just an example)
if(getLocalIp().equals("10.1.2.3"))
{
      //what should i do to disable report?
}
I tried "System.setProperty("skywalking.collector.backend_service","");", but it did not work. The application still reports data to the address written in agent.confg.
My purpose is "same binary + same agent config have different behavior", I do not want to ask someone to keep his agent.config different from others to disable report.

**Comment by wu-sheng** [comment]
https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/setting-override/

Always read document first, please.
